core/trino-main/src/main/java/io/trino/server/TaskResource.java

```diff
@@ -138,11 +142,18 @@ public List<TaskInfo> getAllTaskInfo(@Context UriInfo uriInfo)
     @Produces(MediaType.APPLICATION_JSON)
     public void createOrUpdateTask(
             @PathParam("taskId") TaskId taskId,
+            @QueryParam("nodeId") String requestNodeId,
             TaskUpdateRequest taskUpdateRequest,
             @Context UriInfo uriInfo,
             @Suspended AsyncResponse asyncResponse)
     {
         requireNonNull(taskUpdateRequest, "taskUpdateRequest is null");
+        if (!nodeIdentifier.equals(requestNodeId)) {
+            asyncResponse.resume(Response.status(Status.BAD_REQUEST)
```

Comment:
would it make sense to put this in an Http Filter?

Reply:
would it require to update all the comms requests?

Reply:
should be able to filter by common path if it exists IIRC

Reply:
```
if(path.startsWith(/v1/task) {
//filter logic 
}
```

Reply:
HTTP filter on receiving side would be easy. HTTP client request filter wouldn't be possible, as the filter wouldn't have access to the expected announcement ID for the target.